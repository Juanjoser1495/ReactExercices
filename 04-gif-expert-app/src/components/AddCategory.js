import { useState } from "react";

const AddCategory = () => {
  const [inputValue, setInputValue] = useState("Hello World");
  const handleOnChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleOnSubmit = (e) => {
    e.preventDefault();
    console.log("Submit done");
  };

  return (
    <form onSubmit={handleOnSubmit}>
      <input type="text" value={inputValue} onChange={handleOnChange} />
    </form>
  );
};

export default AddCategory;
