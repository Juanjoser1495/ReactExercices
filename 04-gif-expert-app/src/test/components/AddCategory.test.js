import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Pruebas en <AddCategory />", () => {
  const setCategories = () => {};
  const wrapper = shallow(
    <AddCategory setCategories={setCategories}></AddCategory>
  );
  test("Debe de mostrar el <AddCategory> correctamnete", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
