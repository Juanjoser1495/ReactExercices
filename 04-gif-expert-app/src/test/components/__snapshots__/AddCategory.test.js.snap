// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de mostrar el <AddCategory> correctamnete 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
